### PR TITLE
Added gradient parsing to molcasparser

### PR DESCRIPTION
**Pull request by dpadula85**
Added code to parse gradients in Molcas. I also added the code to parse QM/MM gradients computed with the ESPF method, I left it commented because maybe it is the case to create an additional attribute containing these data.

### CODE REVIEW

**Comment by shivupa** [comment]
Hi Thanks for this and sorry for the delay on the review. This looks great. I have two things. First can you [delete this line](https://github.com/cclib/cclib/blob/1c1671bf09256f615291268c78e2aebb0f27dc35/test/data/testGeoOpt.py#L212) to enable a test for this? Secondly, can you provide more context on the QM/MM ESPF method you mention as I have never used it? When this is used each atom will only have 1 gradient? Or are the qm contributions printed first followed by the mm contributions. The comment you made in the file makes it seem like that isn't the case.  Could you perhaps run the [geometry optimization from the test suite](https://github.com/cclib/cclib/blob/master/data/Molcas/basicOpenMolcas18.0/dvb_gopt.in) with this method to compare or is it too small of a test case.

**Comment by dpadula85** [comment]
Hi, thanks for the suggestions. I removed the line you asked. Before pushing again, I would like to agree with you on the test.
The documentation and examples for the ESPF method, either using the Tinker interface or not, are described here https://www.molcas.org/documentation/manual/node79.html

It's easier if you run the calculation yourself to see what is the output. Basically the program prints the gradients twice: once before the ESPF and once after. The table containing the gradients is the same, but now you have two sets of gradients in the output.
The code I wrote parses gradients for QM calculations in vacuo and only the gradient before the ESPF in QM/MM calculations. The commented code, instead, parses also the gradients after the QM/MM correction.

Since I don't know exactly how you dealt with similar situations for other codes, probably you want to parse both sections and keep them in separate attributes. Or, in case of a QM/MM calculation, store the gradients after QM/MM correction in the main attribute and keep the other ones in a "hidden" attribute (e.g. ``self._grads``).

If you want exactly the same test from the cclib suite, we have to modify it to include the ESPF section. I suppose, otherwise, we can use the test provided on the Molcas website.

Let me know what you prefer, so I can add the test in my branch and repush it.

**Comment by langner** [comment]
Hi @dpadula85 - I don't think you removed that line?

I would recommend first commiting the obvious change - adding the normal gradients. And then proposing another change with the additional gradients, with a recommendation on whether to return multiple gradients or overwrite the previous ones.

**Comment by dpadula85** [comment]
Hi @langner, so I pushed two changes. The first removing the line for the tests. The second is the code to parse QM/MM gradients and overwrite the previous ones, since I think this is preferable.

Thanks for the help!

File: cclib/parser/molcasparser.py [hunk 1]
```diff
@@ -686,6 +686,74 @@ def extract(self, inputfile, line):
...
+            _grads = []
+            line = next(inputfile)
+            while len(line.split()) == 4:
+                tmpgrads = list(map(float, line.split()[1:]))
+                _grads.append(tmpgrads)
+                line = next(inputfile)
+
+            self._grads.append(_grads)
+
+        # Replace the main attribute with QM/MM gradients
```
Comment (langner) [inline cclib/parser/molcasparser.py]: Could you just accumulate into a local variable in the block above, and then do `self.set_attribute('grads', grads)`? That is enough to ensure that the ESPF gradients are parsed and provided when available.

**Comment by langner** [comment]
> Hi @langner, so I pushed two changes. The first removing the line for the tests. The second is the code to parse QM/MM gradients and overwrite the previous ones, since I think this is preferable.

Yup, that's great. Are you able to provide an example logfile with ESPF gradients, which we could add as a regression?

**Comment by langner** [comment]
Note: I think handling this case more generally is similar to other issues we've encountered in the past, for example #99, #657 and #419 most generally. We'll address this as time permits for version 2.x

**Comment by berquist** [comment]
Semi-related, but could you also fix the incorrect docstring here? https://github.com/cclib/cclib/blob/ebf1309e2c4a49d23735689cc7293418dd111c19/test/data/testGeoOpt.py#L251

File: cclib/parser/molcasparser.py (hunk 1 above)
Comment (dpadula85) [reply cclib/parser/molcasparser.py]: I had to do a bit differently to overwrite all previous gradients parsed to be overwritten.

**Comment by berquist** [comment]
@dpadula85 Thanks for your patience!

**Pull request merged**